armv7a9-zynq7000/serial.c

```diff
@@ -0,0 +1,343 @@
+/*
+ * Phoenix-RTOS
+ *
+ * plo - operating system loader
+ *
+ * i.MXRT1064 Serial driver
+ *
+ * Copyright 2020 Phoenix Systems
+ * Author: Hubert Buczyński
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include "peripherals.h"
+#include "zynq.h"
+#include "../serial.h"
+#include "../errors.h"
+#include "../timer.h"
+#include "../low.h"
+
+
+#define MAX_TXRX_FIFO_SIZE  0x40
+#define BUFFER_SIZE         0x200
+
+
+typedef struct {
+	volatile u32 *base;
+	u16 irq;
+
+	u16 rxFifoSz;
+	u16 txFifoSz;
+
+	u8 rxBuff[BUFFER_SIZE];
+	u16 rxHead;
+	u16 rxTail;
+
+	u8 txBuff[BUFFER_SIZE];
+	u16 txHead;
+	u16 txTail;
+
+	u8 tFull;
+} serial_t;
+
+
+enum {
+	cr, mr, ier, idr, imr, isr, baudgen, rxtout, rxwm, modemcr, modemsr, sr, fifo,
```

Comment:
Initialize enum with `cr = 0`. Otherwise, according to the standard, enum can start with any number, picked by the compiler.

Reply:
Done